src/matrices.jl

```diff
@@ -0,0 +1,108 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    JMatrix{T}
+
+Square matrix of ones.
+"""
+struct JMatrix{T} end
+(J::JMatrix{T})(d::Integer) where {T} = Ones{T}(d, d)
+*(::JMatrix, v::AbstractVector) = Fill(sum(v), length(v))
+*(v::Adjoint{<:Any, <:AbstractVector}, J::JMatrix) = (J*v')'
+
+"""
+    FMatrix{T}
+
+`F` matrix, as defined by Aitchison 1986.
+"""
+struct FMatrix{T} end
+(F::FMatrix{T})(d::Integer) where {T} = [Diagonal(fill(one(T), d)) -Ones{T}(d)]
+*(::FMatrix{T}, v::AbstractVector) where {T} = Fill(-last(v), length(v)-1) + v[1:end-1]
+*(v::Adjoint{<:Any, <:AbstractVector}, ::FMatrix) = [v -sum(v)]
+
+"""
+    GMatrix{T}(N)
+
+`G` matrix, as defined by Aitchison 1986.
+"""
+struct GMatrix{T} end
+(G::GMatrix{T})(D::Integer) where {T} = -ones(T, D, D)/D + Diagonal(fill(one(T), D))
+*(::GMatrix, v::AbstractVector) = Fill(-sum(v)/length(v), length(v)) + v
```

Comment:
Another example where we do not benefit from `Fill`? Can we simplify it to `v .- sum(v)/length(v)`

Reply:
Done